EXILED/Exiled.Events/Patches/Generic/IndividualFriendlyFire.cs

```diff
@@ -89,17 +89,18 @@ public static bool CheckFriendlyFirePlayerRules(Footprint attackerFootprint, Ref
 
             // Return false, no custom friendly fire allowed, default to NW logic for FF. No point in processing if FF is enabled across the board.
             if (Server.FriendlyFire)
-                return HitboxIdentity.IsEnemy(attackerFootprint.Role, victimHub.roleManager.CurrentRole.RoleTypeId);
+                return false;
```

Comment:
I am confused, if FriendlyFire has been set to true, why return false?

Reply:
Goes to NW logic

Reply:
This code's behavior is, if true, we are doing custom FF logic, do not use NW logic. If false, we're not using custom ff logic, or we got to the end and we didn't find any rules we could follow, for safety, return false and use NW logic.

Reply:
![image](https://github.com/user-attachments/assets/40e09a90-b756-443b-b8ee-fa439d2cf059)

Reply:
> Goes to NW logic

Well iirc when FriendlyFire is false, the client doesn't even send the damage message to the server when both players are in the same team. So it wouldn't really work?

Reply:
If(Server.FriendlyFire)
-> Return False -> No matter what rules we set, friendly fire has been enabled, we have nothing to bypass - there's no need to worry about X class bypassing role protection on Y class, it's already been enabled. 

I am confused on your confusion:

1. If the friendly fire is set to false, and the client does not send the damage packet then this function is never called; therefore, no need to worry about that scenario. This method presumes there is a packet that comes in and says "I am doing damage to X player, is this allowable, process damage logic path confirm" - If that is not the case, then this entire friendly fire behavior would never work for the same team scenario, since it would never get called. That does not sound right, though I have been away for over a year; therefore, that may be the case. 
2. If the friendly fire is set to true, and the client does send the damage packet, then this function is called, and if it's true then we do not need to worry about what protection rules exist, since it's been bypassed anyway. Therefore, we return back to NW logic which will answer with the same "Oh server wide friendly fire is enabled, go straight to ProcessDamage"

Reply:
My bad for not reading the transpiler part, I read this incorrect documentation which leads to the confusion (also another PR was doing it in the opposite way.)
`/// <returns> True if the attacker can damage the victim.</returns>`

Reply:
Note that `HitboxIdentity.IsDamageable` is not used in Flashbang event, the ProcessEvent method mentions above is used,

Reply:
I see this ExplodingFlashGrenade has been changed - considering options for it

Reply:
> Note that `HitboxIdentity.IsDamageable` is not used in the Flashbang event, the ProcessEvent method mentioned above is used,

What's the reason why it's not used

Reply:
There's 3 option commits, look around and tell me what you think

Reply:
> > Note that `HitboxIdentity.IsDamageable` is not used in the Flashbang event, the ProcessEvent method mentioned above is used,
> 
> What's the reason why it's not used

In that Flashbang patch, the codes for adding targets in base game are skipped

Reply:
The purpose of the CheckFriendlyFire method is not consistent with the patch you were editing previously. 

Instead of return a bool indicating allowing damage, the patch wants a bool for is there a custom FF from what I could tell.

Reply:
ngl without worrying IndividualFriendlyFire rule(still need to look into it later), option 2 looks fine for just solving the flashbang bug to me.

Reply:
> > > Note that `HitboxIdentity.IsDamageable` is not used in the Flashbang event, the ProcessEvent method mentioned above is used,
> > 
> > 
> > What's the reason why it's not used
> 
> In that Flashbang patch, the codes for adding targets in base game are skipped

I can see that, what I meant was, why

Reply:
> Option 2 it is?

if you put option 2 don't use Footprint:Hub but Footprint:Role that very important for things like grenade when you get your rôle changed (i don't really get the difference for them but that very important to not use ReferenceHub here)

Reply:
Done

Reply:
> Done

thanks and even if you choose an other option still does this change :3 because it's needed